src/index.ts

```diff
@@ -261,39 +287,29 @@ export class EmailLinkStrategy<User> extends Strategy<
 
     try {
       // If we get here, the user clicked on the magic link inside email
-      const magicLink = session.get(this.sessionMagicLinkKey) ?? ''
+      const magicLink = session.get('magicLink') ?? ''
       const { emailAddress: email, form } = await this.validateMagicLink(
         request.url,
-        await this.decrypt(magicLink)
+        await this.decrypt(magicLink as string)
       )
       // now that we have the user email we can call verify to get the user
       user = await this.verify({ email, form, magicLinkVerify: true })
     } catch (error) {
-      // if something happens, we should redirect to the failureRedirect
-      // and flash the error message, or just throw the error if failureRedirect
-      // is not defined
-      if (!options.failureRedirect) {
-        throw error
-      }
       const { message } = error as Error
-      session.flash(this.sessionErrorKey, { message })
-      const cookie = await sessionStorage.commitSession(session)
-      throw redirect(options.failureRedirect, {
+      session.flash('error', message)
```

Comment:
any reason you're choosing to flash the message text directly as opposed to within an object?